src/trip/trip.service.ts

```diff
@@ -1,49 +1,78 @@
-import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
 import { Repository } from 'typeorm';
-import { Trip } from "./trip.entity";
+import { Trip } from './trip.entity';
 import { InjectRepository } from '@nestjs/typeorm';
-import { GoogleMapsService } from "../google-maps/google-maps.service";
-import { TravelMode } from "@googlemaps/google-maps-services-js";
-import { pricePerKm } from "../config/price.config";
-import { TripDto } from "./dto/trip.dto";
+import { GoogleMapsService } from '../google-maps/google-maps.service';
+import { TravelMode } from '@googlemaps/google-maps-services-js';
+import { tripConfig } from '../config/trip.config';
+import { TripDto } from './dto/trip.dto';
+import { MonthStatisticInterface } from '../statistic/interface/month-statistic.interface';
+import { WeekStatisticInterface } from '../statistic/interface/week-statistic.interface';
 
 @Injectable()
 export class TripService {
-
   constructor(
     @InjectRepository(Trip)
     private readonly tripRepository: Repository<Trip>,
     private readonly googleMapsService: GoogleMapsService,
   ) {}
 
-
-  async create({ start, finish }: TripDto) {
-
+  async create({ start, finish }: TripDto): Promise<void> {
     const distance = await this.googleMapsService.getCoordinates({
       origins: [start],
       destinations: [finish],
       mode: TravelMode.bicycling,
-    })
+    });
 
     if (!distance) {
-      throw new BadRequestException('Distance not found')
+      throw new BadRequestException('One of two locations not found');
     }
 
     const distanceKm = distance / 1000;
-    const price = pricePerKm.price
-    const tripPrice = +(distanceKm * price).toFixed(2)
-
+    const pricePerKm = tripConfig.pricePerKm;
+    const tripPrice = +((distanceKm * pricePerKm) / 100).toFixed(2);
 
     const trip = await this.tripRepository.create({
       start,
       finish,
       distance: distanceKm,
-      pricePerKm:price,
-      tripPrice: tripPrice
+      pricePerKm: pricePerKm,
+      tripPrice: tripPrice,
     });
+
     if (!trip) {
-      throw new NotFoundException('Trip doesnt exist');
+      throw new NotFoundException('Trip dosent exist');
```

Comment:
throw new NotFoundException('Trip doesn'\t exist');

Reply:
you actually have a point